### PR TITLE
Update BindToGenericAsyncCollector binding rule per Brett's feedback.

### PR DESCRIPTION
**Pull request by MikeStall**
- let the caller have full control over instantiating the collector. 
- merge with type filter

### CODE REVIEW

**Comment by brettsam** [comment]
:shipit:

File: src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs [hunk 1]
```diff
@@ -121,54 +121,22 @@ public IBindingProvider BindToGenericValueProvider<TAttribute>(Func<TAttribute,
         /// Create a binding provider for binding a parameter to an <see cref="IAsyncCollector{T}"/> where T is the user parameter's type. 
         /// </summary>
         /// <typeparam name="TAttribute">type of binding attribute on the user's parameter.</typeparam>
-        /// <typeparam name="TConstructorArgument">The type of the constructor argument.</typeparam>
-        /// <param name="asyncCollectorType">type that implements <see cref="IAsyncCollector{T}"/>. Must have a constructor with exactly 1 parameter of type TConstructorArgument.</param>
-        /// <param name="constructorParameterBuilder">builder function to create an instance of the collector's constructor parameter from a resolved attribute.</param>
+        /// <param name="builder">Function that gets passed (Resolved attribute, 'core' collector type) and instantiates an IAsyncCollector</param>
+        /// <param name="filter">  Optional. type filter, called once at index time. If missing, assume True.</param>
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs]: nit: remove extraneous spaces

**Comment by mathewc** [comment]
:ship:

File: src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs (hunk 1 above)
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs]: got it, thanks.  I thought Style cop usually catches these. 

---
In reply to: [63899361](https://github.com/Azure/azure-webjobs-sdk/pull/727#discussion_r63899361) [](ancestors = 63899361)

**Comment by MikeStall** [comment]
Commited.

**Pull request closed without merging**